### PR TITLE
Test documentation

### LINKED ISSUE #866: Fix sphinx build warnings and test documentation build
See discussion at #864.

### PR DESCRIPTION
**Pull request by dmitriy-serdyuk**
Fix sphinx warnings and add documentation build on Travis.

Also, removes duplicated docs which appeared after #1105.

Fixes #866.

### CODE REVIEW

**Comment by dwf** [comment]
Whoa wtf is going on with coveralls

File: blocks/bricks/bn.py [hunk 1]
```diff
@@ -411,6 +411,7 @@ def _nested_brick_property_setter(self, value, property_name):
                                        property_name='conserve_memory'),
                                partial(_nested_brick_property_setter,
                                        property_name='conserve_memory'))
+    conserve_memory.__doc__ = "Conserve memory."
```
Comment (dwf) [inline blocks/bricks/bn.py]: This line is causing a lot of failures.

**Comment by dmitriy-serdyuk** [comment]
Cool, it seems to be building now. I know, that people are going to complain about our harsh test, but it allows that the documentation is mostly ok (syntax is fine, no repetitions, etc).

**Comment by dmitriy-serdyuk** [comment]
Tests become too long, so I'll separate the documentation tests.

**Comment by dwf** [comment]
LGTM, merge at will

**Pull request merged**